tests/pyccel/test_pyccel.py

```diff
@@ -179,6 +184,7 @@ def compare_pyth_fort_output( p_output, f_output, dtype=float ):
 #------------------------------------------------------------------------------
 def pyccel_test(test_file, dependencies = None, compile_with_pyccel = True,
         cwd = None, pyccel_commands = "", output_dtype = float,
+        compile_dep_with_pyccel = True,
```

Comment:
I'm not sure if this is necessary. I am happy to compile the dependencies with pyccel every time we compile the program with pyccel

Reply:
the dependency in this case is the module containing the functions that takes functions as arguments, compiling it with pyccel will lead to error at shared library generation stage

Reply:
Sorry I said that backwards. I meant:
 I am happy to compile the dependencies with gfortran every time `compile_with_pyccel = False`

Reply:
yes this makes sense